diff --git a/alumnos/59088-juarez-gonzalo/tp2/__main__.py b/alumnos/59088-juarez-gonzalo/tp2/__main__.py
index 11dd26d..9192b45 100644
--- a/alumnos/59088-juarez-gonzalo/tp2/__main__.py
+++ b/alumnos/59088-juarez-gonzalo/tp2/__main__.py
@@ -73,14 +73,18 @@ def w_mmap2file(out_filename):
 if __name__ == "__main__":
     args = parse_args(sys.argv[1:])
 
-    if args["sentido"]:
-        rc_rot = ccw_rc_rot
-    else:
+    if args["rotopt"] == WALSH:
+        rc_rot = walsh_rc_rot
+    elif args["rotopt"] == CW:
         rc_rot = cw_rc_rot
+    else:
+        rc_rot = ccw_rc_rot
 
     in_header = search_fileheader(args["filepath"])
     out_header = header_cp(in_header)
-    swap_rc(out_header)
+
+    if args["rotopt"] != 3:
+        swap_rc(out_header)
 
     out_mmap = mmap.mmap(-1, FILESIZE(out_header))
     out_mmap.write(out_header["content"])
@@ -100,10 +104,12 @@ if __name__ == "__main__":
     for p in pool:
         p.join()
 
-    if args["sentido"]:
-        out_filename = "ccw." + args["filename"]
-    else:
+    if args["rotopt"] == WALSH:
+        out_filename = "walsh." + args["filename"]
+    elif args["rotopt"] == CW:
         out_filename = "cw." + args["filename"]
+    else:
+        out_filename = "ccw." + args["filename"]
 
     w_mmap2file(out_filename)
 
diff --git a/alumnos/59088-juarez-gonzalo/tp2/ccw.dog.ppm b/alumnos/59088-juarez-gonzalo/tp2/ccw.dog.ppm
deleted file mode 100644
index 4cd359c..0000000
Binary files a/alumnos/59088-juarez-gonzalo/tp2/ccw.dog.ppm and /dev/null differ
diff --git a/alumnos/59088-juarez-gonzalo/tp2/cw.dog.ppm b/alumnos/59088-juarez-gonzalo/tp2/cw.dog.ppm
deleted file mode 100644
index 5e22a4a..0000000
Binary files a/alumnos/59088-juarez-gonzalo/tp2/cw.dog.ppm and /dev/null differ
diff --git a/alumnos/59088-juarez-gonzalo/tp2/parse.py b/alumnos/59088-juarez-gonzalo/tp2/parse.py
index ff479dd..0e44faf 100644
--- a/alumnos/59088-juarez-gonzalo/tp2/parse.py
+++ b/alumnos/59088-juarez-gonzalo/tp2/parse.py
@@ -21,18 +21,26 @@ def usagendie():
     sys.stdout.write(h)
     sys.exit(1)
 
+CCW = 1
+CW = 2
+WALSH = 3
+
 def parse_args(argv):
-    opt, args = getopt.getopt(argv, "s:f:h", ["size=", "file=", "sentido", "help"])
+    opt, args = getopt.getopt(argv, "s:f:h", ["size=", "file=", "sentido", "walsh", "help"])
     filename = ""
     filepath = ""
     rsize = 0
-    rotopt = False
+    rotopt = CCW
 
     for o in opt:
         oname = o[0].replace("-","")
 
         if oname == "sentido":
-            rotopt = True
+            rotopt = CW
+            continue
+
+        if oname[0] == "w":
+            rotopt = WALSH
             continue
 
         if oname[0] == "s":
@@ -61,7 +69,7 @@ def parse_args(argv):
         "filename": filename,
         "filepath": filepath,
         "rsize": rsize,
-        "sentido": rotopt
+        "rotopt": rotopt
     }
 
 def search_fileheader(filepath):
diff --git a/alumnos/59088-juarez-gonzalo/tp2/rot.py b/alumnos/59088-juarez-gonzalo/tp2/rot.py
index 268bff4..5620771 100644
--- a/alumnos/59088-juarez-gonzalo/tp2/rot.py
+++ b/alumnos/59088-juarez-gonzalo/tp2/rot.py
@@ -12,6 +12,12 @@ def cw_rc_rot(out_header, in_r, in_c):
     out_c = out_header["cols"] - in_r - 1
     return out_r, out_c
 
+# walsh rot: maria elena walsh - reino del reves
+def walsh_rc_rot(out_header, in_r, in_c):
+    out_r = out_header["rows"] - in_r - 1
+    out_c = out_header["cols"] - in_c - 1
+    return out_r, out_c
+
 # recibe offset en el cuerpo del archivo correspondiente a in_header
 # produce offset en archivo (header + cuerpo) de output
 def byte_rot(rc_rot, in_header, out_header, in_offset):
